src/shogun/classifier/svm/LibLinear.cpp

```diff
@@ -273,6 +283,18 @@ void CLibLinear::solve_l2r_l1l2_svc(
 	double PGmin_old = -CMath::INFTY;
 	double PGmax_new, PGmin_new;
 
+	// for multi-core dual CD
+	// candidates: a block considered for gradient evaluation
+	// workingset: a subset of candidates for sequential CD updates
+	double eps1 = 0.1;
+	double min_eps1 = CMath::min(0.01*eps, eps1);
+	int init_candidates_size = 256;
+	int max_candidates_size = 4096;
+	int candidates_size = CMath::min(init_candidates_size, max_candidates_size);
+	double *Grad = new double[max_candidates_size];
```

Comment:
shogun naming woul dbe lower case, if this is taken from liblinear code, it is fine